routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+
+	// Compute the additional transaction fees as specified
+	// by the request body and the node-level fees.
+	additionalOutputs, err := fes.getTransactionFee(
+		lib.TxnTypeCreateUserAssociation,
+		transactorPublicKeyBytes,
+		requestData.TransactionFees,
+	)
+	if err != nil {
+		_AddBadRequestError(ww, "CreateUserAssociation: specified TransactionFees are invalid")
+		return
+	}
+
+	// Create transaction.
+	txn, totalInput, changeAmount, fees, err := fes.blockchain.CreateCreateUserAssociationTxn(
+		transactorPublicKeyBytes,
+		&lib.CreateUserAssociationMetadata{
+			TargetUserPublicKey: lib.NewPublicKey(targetUserPublicKeyBytes),
+			AppPublicKey:        lib.NewPublicKey(appPublicKeyBytes),
+			AssociationType:     []byte(requestData.AssociationType),
+			AssociationValue:    []byte(requestData.AssociationValue),
+		},
+		extraData,
+		requestData.MinFeeRateNanosPerKB,
+		fes.backendServer.GetMempool(),
+		additionalOutputs,
+	)
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("CreateUserAssociation: problem creating txn: %v", err))
+		return
+	}
+
+	// Construct response.
+	txnBytes, err := txn.ToBytes(true)
+	if err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem encoding txn to bytes")
+		return
+	}
+	res := AssociationTxnResponse{
+		SpendAmountNanos:  totalInput - changeAmount - fees,
+		TotalInputNanos:   totalInput,
+		ChangeAmountNanos: changeAmount,
+		FeeNanos:          fees,
+		Transaction:       txn,
+		TransactionHex:    hex.EncodeToString(txnBytes),
+		TxnHashHex:        txn.Hash().String(),
+	}
+	if err = json.NewEncoder(ww).Encode(res); err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem encoding response as JSON")
+		return
+	}
+}
+
+func (fes *APIServer) DeleteUserAssociation(ww http.ResponseWriter, req *http.Request) {
+	// Decode request body.
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := DeleteAssociationRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, "DeleteUserAssociation: problem parsing request body")
+		return
+	}
+
+	// Create UTXO view.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, "DeleteUserAssociation: problem getting UTXO view")
+		return
+	}
+
+	// Parse TransactorPublicKeyBytes from TransactorPublicKeyBase58Check.
+	if requestData.TransactorPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, "DeleteUserAssociation: must provide a TransactorPublicKeyBase58Check")
+		return
+	}
+	transactorPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		requestData.TransactorPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		_AddInternalServerError(ww, "DeleteUserAssociation: problem getting public key for the transactor")
+		return
+	}
+
+	// Parse AssociationIDBytes from AssociationID (hex string).
+	if requestData.AssociationID == "" {
```

Comment:
should we validate that the association exists and belongs to this transactor?

Reply:
Those validations exists in core. IMO there's some easy validations we should duplicate here (like a string isn't empty) but otherwise these endpoints should just be light wrappers around core functions and we should trust the core validations. Wdyt?